ark/utils/test_utils.py

```diff
@@ -0,0 +1,256 @@
+import os
+from random import choices
+from string import ascii_lowercase
+import numpy as np
+import pandas as pd
+import xarray as xr
+import skimage.io as io
+
+from mibidata import mibi_image as mi, tiff
+
+# required metadata for mibitiff writing (barf)
+MIBITIFF_METADATA = {
+    'run': '20180703_1234_test', 'date': '2017-09-16T15:26:00',
+    'coordinates': (12345, -67890), 'size': 500., 'slide': '857',
+    'fov_id': 'Point1', 'fov_name': 'R1C3_Tonsil',
+    'folder': 'Point1/RowNumber0/Depth_Profile0',
+    'dwell': 4, 'scans': '0,5', 'aperture': 'B',
+    'instrument': 'MIBIscope1', 'tissue': 'Tonsil',
+    'panel': '20170916_1x', 'mass_offset': 0.1, 'mass_gain': 0.2,
+    'time_resolution': 0.5, 'miscalibrated': False, 'check_reg': False,
+    'filename': '20180703_1234_test', 'description': 'test image',
+    'version': 'alpha',
+}
+
+
+def _gen_tif_data(fov_number, chan_number, img_shape, fills, dtype):
+    if not fills:
+        tif_data = np.random.randint(0, 100,
+                                     size=(fov_number, *img_shape, chan_number)).astype(dtype)
+    else:
+        tif_data = np.full(
+            (*img_shape, fov_number, chan_number),
+            (np.arange(fov_number * chan_number) % 256).reshape(fov_number, chan_number),
+            dtype=dtype
+        )
+        tif_data = np.moveaxis(tif_data, 2, 0)
+
+    return tif_data
+
+
+def _gen_label_data(fov_number, comp_number, img_shape, dtype):
+    label_data = np.zeros((fov_number, *img_shape, comp_number), dtype=dtype)
+
+    right = (img_shape[1] - 1) // 2
+    left = (img_shape[1] + 2) // 2
+    up = (img_shape[0] - 1) // 2
+    down = (img_shape[0] + 2) // 2
+
+    counter = 1
+    for fov in range(fov_number):
```

Comment:
This makes quadrants with different values, right?

Reply:
basically yeah.  There's a border of 0's between each quadrant.  Not sure how useful it is, but I figured it was better than all-zeros or random values (which would mess up region props).

Reply:
Ooooh, that's great, having something which will work with regionprops is a great idea